android/src/main/java/com/swmansion/rnscreens/bottomsheet/BottomSheetTransitionCoordinator.kt

```diff
@@ -0,0 +1,135 @@
+package com.swmansion.rnscreens.bottomsheet
+
+import android.os.Build
+import android.view.View
+import android.view.ViewGroup
+import android.view.WindowInsets
+import androidx.annotation.RequiresApi
+import androidx.core.view.WindowInsetsCompat
+import com.swmansion.rnscreens.Screen
+import com.swmansion.rnscreens.ScreenStackFragment
+
+class BottomSheetTransitionCoordinator {
+    private var isLayoutComplete = false
+    private var areInsetsApplied = false
+
+    private var lastInsets: WindowInsets? = null
+    private var lastInsetsCompat: WindowInsetsCompat? = null
```

Comment:
Why store both? Wouldn't it be easier to maintain if we stored only one of them & converted to second ones when needed?